datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1471,7 +1509,10 @@ impl LogicalPlan {
                     // Preserve name to avoid breaking column references to this expression
                     Ok(transformed_expr.update_data(|expr| original_name.restore(expr)))
                 }
-            })
+            })?
+            // always recompute the schema to ensure the changed in the schema's field should be
+            // poplulated to the plan's parent
+            .map_data(|plan| plan.recompute_schema())
```

Comment:
This needs to computed even though plan isn't transformed.
Otherwise changes in a child's schema cannot be populated to
its ancestors.